### PR TITLE
Bullboard auth

### PR DESCRIPTION
**Pull request by VidithM**
For issue #9. Now, whenever the background is running and we go to /bull-board, if we are not authenticated we are first redirected to a login page. The login credentials can be configured in .env.

### CODE REVIEW

**Comment by SysSn13** [comment]
Please remove unnecessary changes. I think you recompiled the node addon.

**Comment by VidithM** [comment]
How about now?

**Comment by SysSn13** [comment]
You have deleted too much stuff, unstage those changes instead of deleting files.

**Comment by SysSn13** [comment]
This is not how it works. You should reset your commits and then stage only desired changes.

**Comment by VidithM** [comment]
Sorry, I am working on it. I'll comment when I get the right version

**Comment by VidithM** [comment]
Ok, I think it should be good now? Let me know.

File: background.js [hunk 1]
```diff
@@ -37,4 +37,6 @@ const initScheduler = async () => {
 
 initScheduler();
 
+serverAdapter.getRouter();
```
Comment (SysSn13) [inline background.js:40]: No use.

File: main.js [hunk 2]
```diff
@@ -32,24 +32,42 @@ app.use(limiter);
...
+app.use(express.static("public"));
+
 // background
 if (process.env.BACKGROUND == true) {
     const { bullBoardServerAdapter } = require("./background");
-    app.use("/bull-board", bullBoardServerAdapter.getRouter());
+    const {ensureLoggedIn} = require("connect-ensure-login");
+    const passport = require('passport');
+    const session = require('express-session');
+    app.use(session({secret : 'keyboard cat'}));
```
Comment (SysSn13) [inline main.js:46]: use SESSION_SECRET env variable for express-session secret.

File: main.js [hunk 3]
```diff
@@ -32,24 +32,42 @@ app.use(limiter);
...
     app.set("layout extractScripts", true);
     app.use(expressLayouts);
-    app.use(bodyParser.urlencoded({ extended: true }));
-    app.use(express.static("public"));
-    app.use("/", webRouter);
+    const unless = require('express-unless');
+    const webRouter = require("./web");
+    webRouter.unless = unless;
+    app.use("/", 
+        webRouter.unless({
```
Comment (SysSn13) [inline main.js:67]: Is this really required?

File: routes/auth.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+passport.serializeUser((user, cb) => {
+    cb(null, user);
+});
+
+passport.deserializeUser((user, cb) => {
+    cb(null, user);
+});
+
+router.get('/', (req, res) => {
+    //console.log('Received login');
```
Comment (SysSn13) [inline routes/auth.js:25]: remove it

File: routes/auth.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+const express = require('express');
+const router = express.Router();
+
+const passport = require('passport');
+const LocalStrategy = require('passport-local').Strategy;
+
+passport.use(
+    new LocalStrategy(function (username, password, cb) {
+      if (username === process.env.BULLBOARD_USERNAME && password === process.env.BULLBOARD_PASS) {
+        return cb(null, { user: 'bull-board' });
```
Comment (SysSn13) [inline routes/auth.js:10]: can we pass BULLBOARD_USERNAME instead of the default 'bull-board' value

File: main.js [hunk 6]
```diff
@@ -32,24 +32,42 @@ app.use(limiter);
...
 }
 
 // web
 if (process.env.WEB == true) {
-    const webRouter = require("./web");
-    app.set("view engine", "ejs");
-    app.set("views", __dirname + "/views");
     app.set("layout", "layouts/layout");
     app.set("layout extractScripts", true);
     app.use(expressLayouts);
```
Comment (SysSn13) [inline main.js:62]: move it top along with other middleware, It is required for login view too

**Review by SysSn13 — CHANGES_REQUESTED** [review]

**Comment by VidithM** [comment]
I also added a 404 page for invalid paths

**Review by SysSn13 — APPROVED** [review]

**Review by SysSn13 — APPROVED** [review]

**Pull request merged**